cfgmgr/natmgr.cpp

```diff
@@ -0,0 +1,6391 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include "logger.h"
+#include "producerstatetable.h"
+#include "macaddress.h"
+#include "natmgr.h"
+#include "exec.h"
+#include "tokenize.h"
+#include "converter.h"
+#include "shellcmd.h"
+#include "warm_restart.h"
+#include "ipaddress.h"
+#include "ipprefix.h"
+
+using namespace std;
+using namespace swss;
+
+/* NatMgr Constructor */
+NatMgr::NatMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, const vector<string> &tableNames) :
+        Orch(cfgDb, tableNames),
+        m_cfgStaticNatTable(cfgDb, CFG_STATIC_NAT_TABLE_NAME),
+        m_cfgStaticNaptTable(cfgDb, CFG_STATIC_NAPT_TABLE_NAME),
+        m_cfgNatPoolTable(cfgDb, CFG_NAT_POOL_TABLE_NAME),
+        m_cfgNatBindingsTable(cfgDb, CFG_NAT_BINDINGS_TABLE_NAME),
+        m_cfgNatGlobalTable(cfgDb, CFG_NAT_GLOBAL_TABLE_NAME),
+        m_cfgInterfaceTable(cfgDb, CFG_INTF_TABLE_NAME),
+        m_cfgLagInterfaceTable(cfgDb, CFG_LAG_INTF_TABLE_NAME),
+        m_cfgVlanInterfaceTable(cfgDb, CFG_VLAN_INTF_TABLE_NAME),
+        m_cfgLoopbackInterfaceTable(cfgDb, CFG_LOOPBACK_INTERFACE_TABLE_NAME),
+        m_cfgNatAclTable(cfgDb, CFG_ACL_TABLE_NAME),
+        m_cfgNatAclRuleTable(cfgDb, CFG_ACL_RULE_TABLE_NAME),
+        m_statePortTable(stateDb, STATE_PORT_TABLE_NAME),
+        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME),
+        m_stateVlanTable(stateDb, STATE_VLAN_TABLE_NAME),
+        m_stateInterfaceTable(stateDb, STATE_INTERFACE_TABLE_NAME),
+        m_appNatTableProducer(appDb, APP_NAT_TABLE_NAME),
+        m_appNaptTableProducer(appDb, APP_NAPT_TABLE_NAME),
+        m_appTwiceNatTableProducer(appDb, APP_NAT_TWICE_TABLE_NAME),
+        m_appTwiceNaptTableProducer(appDb, APP_NAPT_TWICE_TABLE_NAME),
+        m_appNatGlobalTableProducer(appDb, APP_NAT_GLOBAL_TABLE_NAME),
+        m_appNaptPoolIpTable(appDb, APP_NAPT_POOL_IP_TABLE_NAME)
+{
+    SWSS_LOG_ENTER();
+  
+    SWSS_LOG_INFO("NatMgr Constructor ..!");
+
+    /* Set the Admin mode to disabled */
+    natAdminMode = DISABLED;
+
+    /* Set NAT default timeout as 600 seconds */
+    m_natTimeout = NAT_TIMEOUT_DEFAULT;
+    
+    /* Set NAT default tcp timeout as 86400 seconds (1 Day) */
+    m_natTcpTimeout = NAT_TCP_TIMEOUT_DEFAULT;
+
+    /* Set NAT default udp timeout as 300 seconds */
+    m_natUdpTimeout = NAT_UDP_TIMEOUT_DEFAULT;
+
+    /* Clean the NAT iptables */
+    std::string res;
+    const std::string cmds = std::string("") + IPTABLES_CMD + " -F -t nat ";
```

Comment:
Why clear iptables before start?

Reply:
Before the NAT application, wanted to clearing any stale nat iptables rules.